// Copyright 2017 The CrunchyCrypt Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package crunchy;
option java_package = "com.google.security.crunchy.algs.hybrid";

option java_outer_classname = "HybridTestVectorsOuterClass";
option java_multiple_files = true;

message HybridCrypterTestVector {
  bytes private_key = 1;
  bytes public_key = 2;
  bytes plaintext = 3;
  bytes ciphertext = 4;
}

message HybridCrypterTestVectors {
  repeated HybridCrypterTestVector test_vector = 1;
}

message KemTestVector {
  bytes private_key = 1;
  bytes public_key = 2;
  bytes token = 3;
  bytes info = 4;
  bytes derived = 5;
}

message KemTestVectors {
  repeated KemTestVector test_vector = 1;
}
